### PR TITLE
add tests on expected errors for existing kobo post methods

### PR DESCRIPTION
**Pull request by LadaRudnitckaia**
1. I added tests on expected errors in different situations for the `kobo$post`, `kobo$clone_asset`, `kobo$deploy_asset`, `kobo$import_xls_form`.
2. I also did additional testing of these methods with different types of assets that I didn't include to `test-kobo` script, e.g., I checked that surveys can be cloned to survey, block, question, template; blocks to block, question, survey, template; questions to question, survey, template; templates to survey, template. All works properly, except cloning survey and block to question - a block is created instead. Maybe it is something we could report to Kobo's developers. I can also add tests for all these cases to `test-kobo` script if needed.

To sum up, these three methods are tested and seem to work properly (apart from this small issue) and therefore, tickets #33 , #42, #34, #39 can be closed.

### CODE REVIEW

**Review by MKyhos — APPROVED** [review]
LGTM! :)
